types/react-redux/index.d.ts

```diff
@@ -313,18 +316,17 @@ export interface ConnectOptions {
     withRef?: boolean
 }
 
-export interface ProviderProps {
+export interface ProviderProps<A extends Action> {
     /**
      * The single Redux store in your application.
      */
-    store?: Store<any>;
-    children?: ReactNode;
+    store: Store<any, A>;
 }
 
 /**
  * Makes the Redux store available to the connect() calls in the component hierarchy below.
  */
-export class Provider extends React.Component<ProviderProps, {}> { }
+export class Provider<A extends Action> extends Component<ProviderProps<A>> { }
```

Comment:
It might be necessary to add a type parameter default here.

Reply:
It's inferred from `ProviderProps.store`. Maybe I'm missing something, but I don't see a scenario where this would be a problem, without misusing `react-redux`.